neural_compressor/adaptor/pytorch.py

```diff
@@ -4593,35 +4593,86 @@ def teq_quantize(self, model, tune_cfg, dataloader, calib_func):
             if dataloader is None:
                 assert False, "Please provide dataloader or example_inputs for TEQ algorithm."
             try:
-                for idx, (input, label) in enumerate(dataloader):
-                    self.example_inputs = input
+                for idx, (x, label) in enumerate(dataloader):
+                    self.example_inputs = x.to(model.device)
                     break
             except:
-                for idx, input in enumerate(dataloader):
-                    self.example_inputs = input
+                for idx, x in enumerate(dataloader):
+                    self.example_inputs = x.to(model.device)
                     break
 
+        folding = True
         if 'teq_args' in self.recipes:
-            wbits = self.recipes.get('wbits', 4)
-            group_size = self.recipes.get('group_size', 128)
-            sym = self.recipes.get('scheme', False)
-            folding = self.recipes.get('folding', True)
+            folding = self.recipes['teq_args'].get('folding', True)
+            
+        supported_layers = ['Linear']
+        if folding:
+            from .torch_utils.smooth_quant import GraphTrace
+            tg = GraphTrace()
+            absorb_to_layer, _ = tg.get_absorb_to_layer(model, self.example_inputs, supported_layers)
+        else:
+            absorb_to_layer = {}
+            for name, module in model.named_modules():
+                for op_type in supported_layers:
+                    if op_type == str(module.__class__.__name__):
+                        absorb_to_layer[name] = [name]
+
+        if absorb_to_layer is None or absorb_to_layer == {}:
+            logger.warning('No absorb layer is detected, skip TEQ algorithm')
```

Comment:
why skip if no absorb layers? The also should not depend on the folding/fusion.

Reply:
why AWQ use.

Reply:
@xin3he please sync with Kaokao to align the expectation.

Reply:
absorb_to_layer is None should be trace failure, and {} is empty. please dump precise information. Also, better support customized  absorb_to_layer for developers since automatic detection may have some issues.

Reply:
sync with hexin, fixed.